### PR TITLE
Create ESLint rule to discourage usage of navigate() and prefer Link

### LINKED ISSUE #5468: Create ESLint rule to discourage usage of navigate() and prefer Link
We will soon merge #5462 ; I think it would be nice to find a way to enforce this rule, not as strict requirement but as a recommandation across the codebase.
Using navigate() is OK when clicking on button that is doing more than just redirecting to a page (e.g. login out, creating a new record and redirecting to it). But for pure navigation we should avoid using `onClick={() => navigate(...)`  on the component and instead use Link which is more accessible (browser recognizes the link, it can be opened in a new tab, etc.).

As a general principal, it would be nice to always have url/links when it's only a query change and doesn't involve a mutation (e.g. tabs on show page). 

TBC if creating an eslint rule is relevant, maybe not suited for it.

### PR DESCRIPTION
**Pull request by gitstart-twenty**
### Description
Create ESLint rule to discourage usage of navigate() and prefer Link

### Refs
#5468 

### Demo
Capture-2024-05-29-112852
Capture-2024-05-29-112843

Fixes #5468

### CODE REVIEW

**Review by lucasbordeau — APPROVED** [review]
LGTM

File: packages/twenty-front/src/modules/ui/navigation/navigation-drawer/components/NavigationDrawerBackButton.tsx [hunk 1]
```diff
@@ -36,22 +36,19 @@ export const NavigationDrawerBackButton = ({
   title,
 }: NavigationDrawerBackButtonProps) => {
   const theme = useTheme();
-  const navigate = useNavigate();
   const navigationMemorizedUrl = useRecoilValue(navigationMemorizedUrlState);
 
   return (
     <StyledContainer>
-      <StyledIconAndButtonContainer
-        onClick={() => {
```
Comment (FelixMalfait) [inline packages/twenty-front/src/modules/ui/navigation/navigation-drawer/components/NavigationDrawerBackButton.tsx:45]: I think the back button on record show pages was broken and this makes it more evident as it's now a 404

Comment (v1b3m) [reply packages/twenty-front/src/modules/ui/navigation/navigation-drawer/components/NavigationDrawerBackButton.tsx:45]: Yes, please!

Comment (gitstart-twenty) [reply packages/twenty-front/src/modules/ui/navigation/navigation-drawer/components/NavigationDrawerBackButton.tsx:45]: Hey @FelixMalfait 
We've fixed this


https://github.com/twentyhq/twenty/assets/140154534/f90a83ef-cdb3-45fd-b740-0f3117521714


cc @lucasbordeau

**Pull request merged**